### PR TITLE
Add ad738x driver, test script, example code, rst file and emu xml.

### PR DESCRIPTION
**Pull request by spectrum70**
# Description

Add support for the ad738x serie.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?
```

pytest -v --hw=ad7381 --emu --emu-xml=./test/emu/devices/ad7381.xml ./test/test_ad738x.py
================================================================ test session starts =================================================================
platform linux -- Python 3.11.9, pytest-8.2.2, pluggy-1.5.0 -- /home/angelo/dev-baylibre/adi/pyadi-iio/.venv/bin/python3.11
cachedir: .pytest_cache
metadata: {'Python': '3.11.9', 'Platform': 'Linux-6.5.0-rc2-devel-00345-gc2782531397f-x86_64-with-glibc2.39', 'Packages': {'pytest': '8.2.2', 'pluggy': '1.5.0'}, 'Plugins': {'libiio': '0.0.18', 'cov': '5.0.0', 'metadata': '3.1.1', 'html': '3.2.0'}}
rootdir: /home/angelo/dev-baylibre/adi/pyadi-iio
configfile: pyproject.toml
plugins: libiio-0.0.18, cov-5.0.0, metadata-3.1.1, html-3.2.0
collected 1 item                                                                                                                                     

test/test_ad738x.py::test_ad738x_rx_data[1-adi.ad738x] SKIPPED (Test not valid in emulation mode)                                              [100%]

================================================================= 1 skipped in 3.06s =================================================================
Server stopped


invoke precommit
Running supported_parts check
No parts missing from supported_parts.md
---------------

pre-commit run --all-files
codespell................................................................Passed
isort....................................................................Passed
Check Yaml...............................................................Passed
Fix End of Files.........................................................Passed
Trim Trailing Whitespace.................................................Passed
Check for merge conflicts................................................Passed
Debug Statements (Python)................................................Passed
Check docstring is first.................................................Passed
Flake8...................................................................Passed
black....................................................................Passed
blacken-docs.............................................................Passed
Insert license in comments...............................................Passed

example on target  (*)
$ python3 -m ad738x_example
{'voltage0-voltage1': array([-1193.59130785, -1193.18847582, -1193.18847582, ...,
       -1193.18847582, -1193.59130785, -1193.99413988]), 'voltage2-voltage3': array([ 1649.99999898,     0.        ,     0.        , ...,
       -3299.99999795, -1649.99999898,  1649.99999898])}
```
(*)
Example on target requires:
1) to set uri as the target board uri
2) on target board:
cd /sys/bus/iio/devices/iio\:device0/
mkdir /sys/kernel/config/iio/triggers/hrtimer/ad7380_timer
echo ad7380_timer > trigger/current_trigger && iio_attr -d ad7380_timer sampling_frequency 10000

**Test Configuration**:
* Hardware: Host PC, arch linux, target Zynq zed
* OS: arch linux

uImage/devicetree/BOOT.BIN as in 
https://baylibre.atlassian.net/browse/ADI-424

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by spectrum70** [comment]
Hi @tfcollins ,
could you maybe have a look ? Thanks.

File: examples/ad738x_example.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Copyright (C) 2022 Analog Devices, Inc.
+#
+# SPDX short identifier: ADIBSD
+import adi
+import numpy as np
+
+ad738x_dev = adi.ad738x(uri="ip:analog", device_name="ad7381")
+ad738x_dev.rx_annotated = True
+ad738x_dev._rx_data_type = np.int32
```
Comment (tfcollins) [inline examples/ad738x_example.py:9]: This should be automatically determined by the rx methods when the buffer metadata is read. Is that now happening?

**Review by tfcollins — CHANGES_REQUESTED** [review]
Just some monitor tweaks needed

File: test/test_ad738x.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import pytest
+
+hardware = "ad7381"
```
[the remark this answers is not in the document]
Comment (spectrum70) [reply test/test_ad738x.py:3]: Fixed, thanks.

File: examples/ad738x_example.py (hunk 1 above)
Comment (spectrum70) [reply examples/ad738x_example.py:9]: Removed that line, example seems to work properly.

**Comment by spectrum70** [comment]
Hi @tfcollins,

- fixed above points
- re-run all the tests above
```

ytest -v --hw=ad7381 --emu --emu-xml=./test/emu/devices/ad7381.xml ./test/test_ad738x.py
================================================= test session starts ==================================================
platform linux -- Python 3.11.9, pytest-8.2.2, pluggy-1.5.0 -- /home/angelo/dev-baylibre/adi/pyadi-iio/.venv/bin/python3.11
cachedir: .pytest_cache
metadata: {'Python': '3.11.9', 'Platform': 'Linux-6.5.0-rc2-devel-00345-gc2782531397f-x86_64-with-glibc2.40', 'Packages': {'pytest': '8.2.2', 'pluggy': '1.5.0'}, 'Plugins': {'libiio': '0.0.18', 'cov': '5.0.0', 'metadata': '3.1.1', 'html': '3.2.0'}}
rootdir: /home/angelo/dev-baylibre/adi/pyadi-iio
configfile: pyproject.toml
plugins: libiio-0.0.18, cov-5.0.0, metadata-3.1.1, html-3.2.0
collected 1 item                                                                                                       

test/test_ad738x.py::test_ad738x_rx_data[1-adi.ad738x] SKIPPED (Test not valid in emulation mode)                [100%]

================================================== 1 skipped in 3.06s ==================================================

invoke precommit
Running supported_parts check
No parts missing from supported_parts.md
------------------------



pre-commit run --all-files
codespell................................................................Passed
isort....................................................................Passed
Check Yaml...............................................................Passed
Fix End of Files.........................................................Passed
Trim Trailing Whitespace.................................................Passed
Check for merge conflicts................................................Passed
Debug Statements (Python)................................................Passed
Check docstring is first.................................................Passed
Flake8...................................................................Passed
black....................................................................Passed
blacken-docs.............................................................Passed
Insert license in comments...............................................Passed
```

Thanks

**Pull request merged**